pkg/controller/kogitoapp/kogitoapp_controller.go

```diff
@@ -340,29 +342,38 @@ func (r *ReconcileKogitoApp) ensureKogitoImageStream(instance *v1alpha1.KogitoAp
 	return r.createImageStream(instance, "")
 }
 
-func (r *ReconcileKogitoApp) ensureKogitoInfra(instance *v1alpha1.KogitoApp, runtimeImage *docker10.DockerImage, requestedDeployment *oappsv1.DeploymentConfig) (requeue bool, err error) {
-	requeueInfinispan, err := r.ensureInfinispan(instance, runtimeImage, requestedDeployment)
+func (r *ReconcileKogitoApp) ensureKogitoInfra(instance *v1alpha1.KogitoApp, appProps map[string]string) (requeue bool, err error) {
+	requeueInfinispan, err := r.ensureInfinispan(instance, appProps)
 	if err != nil {
 		return false, err
 	}
 
-	requeueKafka, err := r.ensureKafka(instance, requestedDeployment)
+	requeueKafka, err := r.ensureKafka(instance, appProps)
 	if err != nil {
 		return false, err
 	}
 
 	return requeueInfinispan || requeueKafka, nil
 }
 
-func (r *ReconcileKogitoApp) ensureInfinispan(instance *v1alpha1.KogitoApp, runtimeImage *docker10.DockerImage, requestedDeployment *oappsv1.DeploymentConfig) (requeue bool, err error) {
+func (r *ReconcileKogitoApp) ensureInfinispan(instance *v1alpha1.KogitoApp, appProps map[string]string) (requeue bool, err error) {
 	log.Debug("Verify if we need to deploy Infinispan")
-	if instance.Spec.EnablePersistence || framework.IsPersistenceEnabled(runtimeImage) {
+	if instance.Spec.EnablePersistence || framework.IsPersistenceEnabled(instance.Name, instance.Namespace, r.client) {
 		infra, ready, err := infrastructure.EnsureKogitoInfra(instance.Namespace, r.client).WithInfinispan().Apply()
 		if err != nil {
 			return true, err
 		}
 		if ready {
-			if err := kogitores.SetInfinispanEnvVars(r.client, infra, instance, requestedDeployment); err != nil {
+			envs, aps, err := kogitores.CreateInfinispanProperties(r.client, infra, instance)
+			if err != nil {
+				return true, err
+			}
+			envMap := framework.EnvVarArrayToEnvVarMap(instance.Spec.Envs)
```

Comment:
Can this be replaced by `EnvOverride`?  https://github.com/kiegroup/kogito-cloud-operator/blob/master/pkg/framework/env_var.go#L67